actions/maven_compute_release_version/action.yml

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2022 AVI-SPL Inc.
+# All Rights Reserved.
+#
+# Compute Maven project version for a release
+# When user provides a version, use it
+# When user does not provide a version, use current project version with
+# an updated build number. Supporter current version formats are :
+#   - snapshot version (e.g. "5.6.1-SNAPSHOT")
+#   - version with already a build number (e.g. "5.6.1.2208051600")
+# WARNING
+#   For unknown reason the "result" output created by the actions/github-script@v6
+#   action when returning value at the end of script is not visible to the outside.
+#   So explicitly set an output (here, "release-version") which is working fine
+#
+name: 'Maven compute version for release'
+description: 'Maven compute version for release'
+inputs:
+  user-version:
+    description: 'Version provided by user'
+    required: true
+  current-version:
+    description: 'Current project version'
+    required: true
+  build-number:
+    description: 'The build number to be used in produced version (if adequate)'
+    required: true
+outputs:
+  release-version:
+    description: >
+      Project version, e.g. "5.6.1.2208051600"
+    value: ${{ steps.version.outputs.release-version }}
+
+runs:
+  using: 'composite'
+  steps:
+    - id: version
+      uses: actions/github-script@v6
+      with:
+        script: |
+          // Regular expressions for possible versions formats
+          // 5.6.1-SNAPSHOT
+          const REGEXP_SNAPSHOT = /^[0-9]+\.[0-9]+\.[0-9]+-SNAPSHOT$/;
+          // 5.6.1.2208051600
+          const REGEXP_WITH_BUILD_NUMBER = /^[0-9]+\.[0-9]+\.[0-9]+\.([0-9]){10}$/;
+
+          // User version - trim whitespaces (left and right)
+          let user_version = "${{ inputs.user-version }}".trim();
+
+          // Computed version
+          let release_version;
+
+          if (user_version.length === 0) {
```

Comment:
what is difference between `==` and `===`? (just curious)

Reply:
This is Javascript operator. The `===` is an exact equality (compares both value and type) while `==` does type conversion before comparing. So `0 == "0"` is true while `0 === "0"` is false.
In the case of `user_version.length === 0` both would give same result, but the stricter the better